controller/model/authenticator_manager.go

```diff
@@ -599,6 +601,19 @@ func (self *AuthenticatorManager) VerifyExtendCertForIdentity(isJwtBacked bool,
 	authenticatorCert.Pem = authenticatorCert.UnverifiedPem
 	authenticatorCert.Fingerprint = authenticatorCert.UnverifiedFingerprint
 
+	newPublicKeyPrint := PublicKeySha256(verifyCerts[0])
+
+	if authenticatorCert.PublicKeyPrint == "" {
+		//if we don't have a saved value, parse and calculate
+		certs := nfpem.PemStringToCertificates(authenticatorCert.Pem)
+		if len(certs) > 0 {
+			authenticatorCert.PublicKeyPrint = PublicKeySha256(certs[0])
+		}
+	}
+
+	authenticatorCert.LastExtendRolledKeys = newPublicKeyPrint != authenticatorCert.PublicKeyPrint
```

Comment:
PublicKeySha256 can return "" when it fails. In the unlikely case that line 604 or 610 fail this could be true for the wrong reason, right?

Reply:
The error result is improbable. It is because PublicKey is of type  `any`  and can fail type asserting when marshalling. At this point, the cert has been parsed and used multiple times. So it is improbable that it will error.